src/session.ts

```diff
@@ -0,0 +1,133 @@
+import * as superagent from 'superagent';
+
+import debugModule from 'debug';
+const debug = debugModule('everyauth:profile');
+
+import { getAuthedProfile, IProfile } from './profile';
+import EveryAuthVersion from './version';
+import { getInstallIdByTags } from './install';
+
+import { SERVICE_TAG, USER_TAG, TENANT_TAG } from './constants';
+
+const COMMIT_URL_SUFFIX = '/commit';
+
+const META_INTEGRATION_ID = 'everyauth';
+
+export interface ISession {
+  id: string;
+  userId: string;
+  tenantId?: string;
+
+  tags: Record<string, string | null>;
```

Comment:
Why aren't userId and tenantId just part of `tags` at this level?

Reply:
This was some debris to make error handling a smidge easier - refactored to keep it closer to the actual usage.